### ISSUE TITLE
self.local_server_settings mismatch with HTTPS and HTTP

### ISSUE THREAD

**Issue by paulie4** [SalesforceFoundation/salesforce-requests-oauthlib#5]
I don't understand how `self.local_server_settings` is compatible with both of the following blocks of code.

[This code](https://github.com/SalesforceFoundation/salesforce-requests-oauthlib/blob/master/src/salesforce_requests_oauthlib/__init__.py#L116) has a hard-coded "https://":
```
        self.callback_url = 'https://{0}:{1}'.format(
            self.local_server_settings[0],
            str(self.local_server_settings[1])
        )
```

but [this code](https://github.com/SalesforceFoundation/salesforce-requests-oauthlib/blob/master/src/salesforce_requests_oauthlib/__init__.py#L210), which creates a `BaseHTTPServer.HTTPServer`, which according to [the docs](https://docs.python.org/2/library/basehttpserver.html), creates an HTTP socket, not an HTTPS socket:
```
        httpd = BaseHTTPServer.HTTPServer(
            self.local_server_settings,
            RequestHandler
        )
```

**Comment by paulie4** [comment]
Oh, I just realized that if you configure the Callback URL for a connected app in Salesforce to use http://localhost, it'll redirect to http://localhost, even if the &redirect_uri= in the https://login.salesforce.com/services/oauth2/authorize request says https://localhost, so that's why this is not really a problem, but the code should probably have a comment saying why that's OK.

**Comment by adamlincoln** [comment]
@paulie4 good catch, I like your proposed comment.  If you feel like opening a PR I'll merge it, or I'll throw it into the next release when it comes along.  Thank you!

**Comment by adamlincoln** [comment]
Closed by PR #6 , thanks @paulie4 .
